pwa/src/components/DiagnosisForm.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { ReactElement } from 'react'
+import { List, Checkbox, InputItem, WhiteSpace } from 'antd-mobile'
+import '../scss/DiagnosisForm.scss'
+import * as data from './DiagnosisFields.json'
+
+/*
+
+Diagnosis Fields are stored in DiagnosisFields.json in the format:
+
+{
+    "diagnoses" : [
+        {
+            "name": "Name",
+            "id": "id",
+            "text-input-title": "Title",
+            "text-input-id": "id" for `id` and `data-testid` attributes
+            "text-input-type": "type" (i.e. number)
+            "max-length": 3
+            "group-title": "Group Title",
+            "group": [
+                { "name": "Name1", "id": "id1" },
+                { "name": "Name2", "id": "id2" }
+            ]
+        },
+        {...}
+    ]
+}
+*/
+
+const CheckboxItem = Checkbox.CheckboxItem
+const diagnosisFields = data.diagnoses
+
+export default function DiagnosisForm(): ReactElement {
+    return (
+        <div id="diagnosis-form-wrapper" data-testid="diagnosis-form-wrapper">
+            {diagnosisFields.map((field) => {
+                const formElements: ReactElement[] = []
+
+                if (field.name != null) {
```

Comment:
Checkboxes that do not belong to a `group` structure must be clicked on the circle where the check goes, but checkboxes that are part of a `group` can be checked by clicking anywhere in the larger div.

Is it possible to expand the active input area for the individual checkboxes to include the entire div?

Not sure if that makes sense, but if it doesn't make sense just give me a call.

Reply:
This is resolved [here](https://github.com/Haiti-Distributed-Healthcare-System/hdhs/pull/84/commits/2dcac548c332c4d95e1b5805fc5233d924bbcdd1), I chose to wrap all of the checkboxes that are not part of a group in a List element without a header. This gives them the exact styling of the list elements and changes nothing visually.